src/Extension/InitData.re

```diff
@@ -12,6 +14,22 @@ module Extension = {
     extensionKind: string,
     enableProposedApi: bool,
   };
+
+  let ofScanner: Scanner.t => t =
+    ({manifest, path, _}: Scanner.t) => {
+      // TODO: Is identifier right?
+      identifier: manifest.name,
+      extensionLocation: path |> Uri.fromPath,
+      name: manifest.name,
+      main: manifest.main,
+      version: manifest.version,
+      engines: manifest.engines,
+      activationEvents: manifest.activationEvents,
+      extensionDependencies: manifest.extensionDependencies,
+      // TODO: Convert correctly
+      extensionKind: "ui",
+      enableProposedApi: manifest.enableProposedApi,
+    };
```

Comment:
Likewise, the name `ofScanner` here makes no sense.

Reply:
I changed this to be `ofManifestAndPath` - so not coupled to the scanner types.